File: Session7/C++/TwoSum.cpp
https://leetcode.com/problems/two-sum/

class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        
        unordered_map<int, int> m;
        int n = nums.size();
        
        for(int i=0; i<n; i++)
        {
            int diff = target - nums[i];
            if(m.count(diff) > 0)
            {
                return {i, m[diff]};
            }
            m[nums[i]] = i;
        }
        
        return {};
        
    }
};

File: Session7/Java/TwoSum.java
https://leetcode.com/problems/two-sum/

class Solution {
    public int[] twoSum(int[] nums, int target) {
        
        Map<Integer, Integer> mp = new HashMap<>();
        int n = nums.length;
        
        for(int i=0; i<n; i++)
        {
            int diff = target - nums[i];
            if(mp.containsKey(diff))
            {
                int[] res = {mp.get(diff), i}; 
                return res;
            }
            mp.put(nums[i], i); 
        }
        
        return null;
        
    }
}